Jetpack Cloud: Get Scan status from API

rcanepa opened the pull request:
#### Changes proposed in this Pull Request

- Create a Query Component to fetch the Scan status from the API and load the data into the store.

#### Testing instructions

- Apply the "Add dummy scan api endpoint" patch to your sandbox (D41270-code)
- Navigate to http://calypso.localhost:3000
- http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=threats
- See in the network tab the request to `/site/<siteId>/scan`
- Check the JETPACK_SCAN_REQUEST action has been dispatched
- Check in Redux that the data is stored in `jetpackScan.scan` and the request status in `jetpackScan.requestStatus`

See 1151678672052943-as-1169562672652097.

ChaosExAnima commented on client/state/data-getters/index.js:
```diff
@@ -230,3 +230,33 @@ export const requestSiteAlerts = siteId => {
+					requestId,
+					{
+						...rest,
+						startTimestapm: start_timestamp,
```
typo :)

rcanepa replied:
Opps... Good catch!

rcanepa commented:
> Should this have tests?

I think so. I'll add some.

enejb commented on client/state/data-getters/index.js:
```diff
@@ -230,3 +230,33 @@ export const requestSiteAlerts = siteId => {
+			{}
+		),
+		{
+			freshness: 5 * 60 * 1000,
```
Is this the same as 5min?

rcanepa replied:
We no longer need this data-getter because we are using the `QueryJetpackScan` component to fetch the data and load it into the store.

enejb commented:
This worked really well for me! 

Nicely done. I think after we add some test we should be ready to ship.

rcanepa commented:
I added a couple of test around the `QueryJetpackScan` component. Also, I added tests to verify the fetch handler (data-layer) is getting called when dispatching the `requestJetpackScanStatus` action.

enejb commented:
In my testing this worked as expected. 
Lets ship this and then integrate it with on the Scanner page.
